source/js/buyers-guide/components/social-share/product-quiz-share-buttons.jsx

```diff
@@ -0,0 +1,108 @@
+import React, { Component } from "react";
+import copyToClipboard from "../../../copy-to-clipboard";
+
+class ProductQuizShareButtons extends Component {
+  shareButtonClicked(event, shareProgressButtonId) {
+    if (shareProgressButtonId) {
+      const shareProgressButton = document.querySelector(
+        `#${shareProgressButtonId} a`
+      );
+
+      if (shareProgressButton) {
+        shareProgressButton.click();
+      }
+    } else {
+      // Updating all "copy" buttons (mobile and desktop) to the updated "copied" state.
+      const linkButtons = document.querySelectorAll(".link-share");
+
+      linkButtons.forEach((linkButton) => {
+        linkButton.classList.add("copied");
+        linkButton.title = window.gettext("Copied");
+        const spanElement = linkButton.querySelector("span");
+        spanElement.innerText = window.gettext("Copied");
+      });
+
+      copyToClipboard(event.target, window.location.href);
+    }
+  }
+
+  render() {
+    const shareButtonClasses =
+      "tw-btn tw-bg-white tw-h-22 tw-w-22 tw-border-black hover:tw-bg-black hover:tw-text-white btn-share after:tw-hidden";
+
+    const mobileShareButtons = (
+      <div className="circle share-button-group tw-justify-center tw-flex medium:tw-hidden">
+        <div className="subgroup">
+          <button
```

Comment:
Since the `<button>` on mobile and desktop version are almost identical except their inner element, what do you think about creating a function to render them? Something like

```jsx
renderFacebookButton(child) {
    return (
      <button
        className={`${this.shareButtonClasses} facebook-share`}
        onClick={(e) => this.shareButtonClicked(e, "share-progress-fb")}
      >
        {child}
      </button>
    );
  }
```

And you should be able to build the button group by calling `this.renderFacebookButton(<span className="sr-only">Share on Facebook</span>)`, for example.

Reply:
Great idea Mavis! I have updated this PR to do the following and re-requested review 👍